tests/python_client/testcases/test_utility.py

```diff
@@ -1982,3 +1983,314 @@ def test_delete_user_root(self, host, port):
+        self.utility_wrap.create_role()
+        self.utility_wrap.role_drop()
+        assert not self.utility_wrap.role_is_exist()[0]
+
+    @pytest.mark.tags(CaseLabel.L3)
+    def test_add_user_to_role(self, host, port):
+        """
+        target: test add user to role
+        method: create a new user,add user to role
+        expected: assert add user success
+        """
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+        user = cf.gen_unique_str(prefix)
+        password = cf.gen_unique_str(prefix)
+        r_name = cf.gen_unique_str(prefix)
+        u, _ = self.utility_wrap.create_user(user=user, password=password)
+
+        self.utility_wrap.init_role(r_name)
+        self.utility_wrap.create_role()
+        self.utility_wrap.role_add_user(user)
+        users, _ = self.utility_wrap.role_get_users()
+        user_info, _ = self.utility_wrap.list_user(user, True)
+        user_item = user_info.groups[0]
+        assert r_name in user_item.roles
+        assert user in users
+
+    @pytest.mark.tags(CaseLabel.L3)
+    def test_remove_user_from_role(self, host, port):
+        """
+        target: test remove user from role
+        method: create a new user,add user to role, remove user from role
+        expected: assert remove user from role success
+        """
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+        user = cf.gen_unique_str(prefix)
+        password = cf.gen_unique_str(prefix)
+        r_name = cf.gen_unique_str(prefix)
+        u, _ = self.utility_wrap.create_user(user=user, password=password)
+
+        self.utility_wrap.init_role(r_name)
+        self.utility_wrap.create_role()
+        self.utility_wrap.role_add_user(user)
+        self.utility_wrap.role_remove_user(user)
+        users, _ = self.utility_wrap.role_get_users()
+        assert len(users) == 0
+
+    @pytest.mark.tags(CaseLabel.L3)
+    def test_role_is_exist(self, host, port):
+        """
+        target: test role is existed
+        method: check not exist role and exist role
+        expected: assert is_exist interface is correct
+        """
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+        r_name = cf.gen_unique_str(prefix)
+        r_not_exist = cf.gen_unique_str(prefix)
+        self.utility_wrap.init_role(r_name)
+        self.utility_wrap.create_role()
+        assert self.utility_wrap.role_is_exist()[0]
+        self.utility_wrap.init_role(r_not_exist)
+        assert not self.utility_wrap.role_is_exist()[0]
+
+    @pytest.mark.tags(CaseLabel.L3)
+    def test_role_grant_collection_insert(self, host, port):
```

Comment:
The scenario that insert without granting "insert" privilege will be covered in the next pr covering negative cases?